File: python-algorithms/src/binary_search.py
from typing import List

class BinarySearch:
    def search(self, nums: List[int], target: int) -> int:
        pointer_a = 0
        pointer_b = len(nums) -1

        while pointer_a <= pointer_b:
            mid_point = (pointer_a + pointer_b) // 2
            
            if nums[mid_point] == target:
                return mid_point
            elif nums[mid_point] > target:
                pointer_b = mid_point - 1  
            else:
                pointer_a = mid_point + 1
                
        return -1

File: java-algorithms/src/main/java/com/omsprog/algorithms/lc/BinarySearch.java
package com.omsprog.algorithms.lc;

public class BinarySearch {
    public int search(int[] nums, int target) {
        int pointerA = 0;
        int pointerB = nums.length - 1;
        int midPoint = 0;

        while(pointerA <= pointerB) {
            midPoint = (pointerA + pointerB) / 2;
            int val = nums[midPoint];

            if(val == target) return midPoint;
            else if(val > target)
                pointerB = midPoint - 1;
            else
                pointerA = midPoint + 1;
        }

        return -1;
    }
}
